web/src/components/njwds-extended/UnStyledButton.tsx

```diff
@@ -1,87 +1,46 @@
-    let style = "";
+    const style =
+      props.style === "default"
+        ? "usa-button usa-button--unstyled width-auto font-weight-inherit font-size-inherit"
+        : "usa-button bg-transparent";
     const disabledClass = "usa-button--disabled";
     const widthRef = useRef<HTMLInputElement | null>(null);
     const [width, setWidth] = useState<number>();
     const [height, setHeight] = useState<number>();
 
-    switch (props.style) {
-      case "light":
-        style =
-          "usa-button bg-transparent text-normal text-base border-1px border-base-light hide-unhide-button padding-y-11px";
-        break;
-      case "tertiary":
-        style = "usa-button usa-button--unstyled width-auto font-weight-inherit font-size-inherit";
-        break;
-      case "narrow-light":
-        style =
-          "usa-button usa-tag bg-transparent text-normal text-base border-1px border-base-light hide-unhide-button";
-        break;
-      case "unstyled":
-        style = "usa-button bg-transparent";
-        break;
-    }
-    const showDisabledClass = props.loading ? disabledClass : "";
-
     useEffect(() => {
-      if (!props.loading && widthRef.current !== null) {
+      if (widthRef.current !== null) {
         setWidth(widthRef.current.clientWidth);
         setHeight(widthRef.current.clientHeight);
       }
-    }, [height, width, props.loading, disabledClass]);
+    }, [height, width, disabledClass]);
 
-    const getRightMargin = (): string => {
-      if (props.style === "tertiary") {
-        return "margin-right-0";
-      }
-      return props.noRightMargin ? "margin-right-0" : "margin-right-2";
-    };
-    const heightAutoOnMobile = props.heightAutoOnMobile ? "height-auto" : "";
-    const widthAutoOnMobile = props.widthAutoOnMobile ? "width-auto" : "";
-    const noRightMargin = getRightMargin();
-    const underline = props.underline ? "underline" : "";
-    const smallText = props.smallText ? "font-body-2xs" : "";
-    const textBold = props.textBold ? "text-bold" : "";
-    const intercomButton = props.intercomButton ? "intercom-button" : "";
-    const fullWidth = props.fullWidth ? "width-100" : "";
+    const noRightMargin =
+      props.isRightMarginRemoved || props.style === "default" ? "margin-right-0" : "margin-right-2";
+    const underline = props.isUnderline ? "underline" : "";
+    const smallText = props.isSmallerText ? "font-body-2xs" : "";
+    const textBold = props.isTextBold ? "text-bold" : "";
+    const intercomButton = props.isIntercomEnabled ? "intercom-button" : "";
 
-    const className = [
-      props.className,
-      style,
-      heightAutoOnMobile,
-      widthAutoOnMobile,
-      noRightMargin,
-      underline,
-      smallText,
-      textBold,
-      intercomButton,
-      showDisabledClass,
-      fullWidth,
-    ]
+    const className = [style, props.className, noRightMargin, underline, smallText, textBold, intercomButton]
```

Comment:
[Pebble] I think (but am not sure) that in this current version right now, the props based booleans will override the props.className and the style attributes. 

For example if I have className={"font-size-10"} and I also pass isSmallText={true} I think in the current version the isSmallText will override the class but I think in general the className should override the booleans.

Easy resolve by moving the className to the bottom of the list because last in wins (probably should double check that). Thoughts?